pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -164,8 +165,13 @@ func getClassInfos() map[gateway.GatewayController]classInfo {
 // The controller will not start until Run() is called.
 func NewDeploymentController(client kube.Client, clusterID cluster.ID, env *model.Environment,
 	webhookConfig func() inject.WebhookConfig, injectionHandler func(fn func()), tw revisions.TagWatcher, revision string,
+	nsFilter namespace.DiscoveryNamespacesFilter,
 ) *DeploymentController {
-	gateways := kclient.New[*gateway.Gateway](client)
+	var filter namespace.DiscoveryFilter
```

Comment:
please gate by EnableEnhancedResourceScoping

Reply:
good catch, will add the caller